agent/grpc-internal/resolver/resolver.go

```diff
@@ -93,7 +93,7 @@ func (s *ServerResolverBuilder) Build(target resolver.Target, cc resolver.Client
 	}
 
 	//nolint:staticcheck
-	serverType, datacenter, err := parseEndpoint(target.Endpoint)
+	serverType, datacenter, err := parseEndpoint(target.Endpoint())
```

Comment:
This and below one-line changes fix binary incompatibilities introduced by the upgrade; these match later release branches and `main`